usaspending_api/disaster/v2/views/agency/spending.py

```diff
@@ -121,13 +113,35 @@ def total_queryset(self):
                 ),
                 0,
             ),
+        }
+
+        cte = With(
+            FinancialAccountsByProgramActivityObjectClass.objects.filter(*cte_filters)
+            .values("treasury_account__funding_toptier_agency_id")
+            .annotate(**cte_annotations)
+            .values(*cte_annotations)
+        )
+
+        annotations = {
+            "id": Subquery(
+                Agency.objects.filter(toptier_agency_id=OuterRef("toptier_agency_id"))
+                .order_by("-toptier_flag", "id")
+                .values("id")[:1]
+            ),
+            "code": F("toptier_code"),
+            "description": F("name"),
+            # Currently, this endpoint can never have children w/o type = `award` & `award_type_codes`
+            "children": Value([], output_field=ArrayField(IntegerField())),
+            "count": Value(0, output_field=IntegerField()),
+            "obligation": cte.col.obligation,
+            "outlay": cte.col.outlay,
```

Comment:
Does calling cte twice here make two ctes? `With` looks like a class instead of a function, so I think this is okay?

Reply:
CTEs are treated like a model, so it's just referencing the attributes in the CTEs rather than executing. Unlike sub-queries which are separate queries executed independently